multiqc/modules/nanoq/nanoq.py

```diff
@@ -0,0 +1,303 @@
+                    thresholds["Percentage"].append(float(percentage))
+            return thresholds
+
+        # Parse the file content
+        segment = None
+        summary_lines = []
+        length_threshold_lines = []
+        quality_threshold_lines = []
+
+        for line in f["f"]:
+            line = line.strip()
+            if line.startswith("Nanoq Read Summary"):
+                segment = "summary"
+                continue
+            elif line.startswith("Read length thresholds"):
+                segment = "length_thresholds"
+                continue
+            elif line.startswith("Read quality thresholds"):
+                segment = "quality_thresholds"
+                continue
+
+            if segment == "summary":
+                summary_lines.append(line)
+            elif segment == "length_thresholds":
+                length_threshold_lines.append(line)
+            elif segment == "quality_thresholds":
+                quality_threshold_lines.append(line)
+
+        # Parse each part
+        nanoq_summary = parse_summary(summary_lines)
+        read_length_thresholds = parse_thresholds(length_threshold_lines, "bp")
+        read_quality_thresholds = parse_thresholds(quality_threshold_lines, "")
+
+        # Prepare the final nanoq stats dictionary
+        nanoq_stats: Dict[str, float] = {}
+        for metric, value in zip(nanoq_summary["Metric"], nanoq_summary["Value"]):
+            if metric in self._KEYS_MAPPING:
+                nanoq_stats[self._KEYS_MAPPING[metric]] = value
+
+        for threshold, num_reads in zip(read_length_thresholds["Threshold"], read_length_thresholds["Number of Reads"]):
+            nanoq_stats[f"Reads > {threshold}"] = num_reads
+
+        for threshold, num_reads in zip(
+            read_quality_thresholds["Threshold"], read_quality_thresholds["Number of Reads"]
+        ):
+            nanoq_stats[f"Reads > Q{threshold}"] = num_reads
+
+        # Store nanoq_summary for general stats
+        self.nanoq_summary[f["s_name"]] = {k: v for k, v in zip(nanoq_summary["Metric"], nanoq_summary["Value"])}
+
+        self.save_data(f, nanoq_stats)
+
+    def save_data(self, f, nanoq_stats: Dict[str, float]) -> None:
+        """
+        Normalise fields and save parsed data.
+        """
+        if any(k.startswith("Reads > Q") for k in nanoq_stats):
+            stat_type = "quality"
+        else:
+            stat_type = "summary"
+
+        out_d = {f"{k}_{stat_type}": v for k, v in nanoq_stats.items()}
+
+        # Warn if we find overlapping data for the same sample
+        if f["s_name"] in self.nanoq_data:
+            if not set(self.nanoq_data[f["s_name"]].keys()).isdisjoint(out_d.keys()):
+                log.debug(f"Duplicate sample data found! Overwriting: {f['s_name']}")
+
+        self.nanoq_data.setdefault(f["s_name"], {}).update(out_d)
+
+        self.add_data_source(f)
+
+    def nanoq_general_stats(self) -> None:
+        """Nanoq General Stats Table"""
```

Comment:
There are too many general stats. If other module are run on a project, the table will end up too wide. Please keep 1 or 2, plus up to 5-7 hidden metrics.

Reply:
Or add a separate table in the report